pkg/api/operators/apis/studyjob/v1alpha1/studyjob_types.go

```diff
@@ -0,0 +1,182 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//runtime "k8s.io/apimachinery/pkg/runtime"
+
+	pb "github.com/kubeflow/katib/pkg/api"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// StudyJobSpec defines the desired state of StudyJob
+type StudyJobSpec struct {
+	StudySpec            *StudySpec            `json:"studySpec,omitempty"`
```

Comment:
The name StudySpec seems confusing; I would think the whole thing is the StudySpec.

What's the distinction between StudySpec and SuggestionSpec?

Reply:
SuggestionSpec is parameters that related to only suggestion service.
How about merge StudySpec to StudyJobSpec like below.

```
type StudyJobSpec struct {
	Name               string            `json:"name,omitempty"`
	Owner              string            `json:"owner,omitempty"`
	OptimizationType   OptimizationType  `json:"optimizationtype,omitempty"`
	OptimizationGoal   *float64          `json:"optimizationgoal,omitempty"`
   ....
	WorkerSpec           *WorkerSpec           `json:"workerSpec,omitempty"`
	SuggestionSpec       *SuggestionSpec       `json:"suggestionSpec,omitempty"`
	EarlyStoppingSpec    *EarlyStoppingSpec    `json:"earlyStoppingSpec,omitempty"`
	MetricsCollectorSpec *MetricsCollectorSpec `json:"metricsCollectorSpec,omitempty"`
}
```

Reply:
This looks good to me.